base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/generation/GeneratorPython.java

```diff
@@ -78,6 +81,16 @@ public void refreshTimeEstimation(List<ScriptGeneratorAction> scriptGenContent,
             LOG.error(e);
         }
 	}
+	
+	@Override
+    public void refreshCustomEstimation(List<ScriptGeneratorAction> scriptGenContent, ScriptDefinitionWrapper scriptDefinition, List<String> globalParams) throws InterruptedException, ExecutionException {
+		try {
+			pythonInterface.refreshCustomEstimation(scriptGenContent, scriptDefinition, globalParams);
+		} catch (PythonNotReadyException e) {
+			// ScriptGeneratorSingleton is listening to python interface readiness changes (handled there)
+            LOG.error(e);
+		}
+	}
```

Comment:
Once again, we have lots of methods with the same boilerplate, so I'm going to suggest the same sort of refactoring as above...